pkg/extract/redhat/csaf/csaf.go

```diff
@@ -680,216 +687,354 @@ func walkVulnerabilities(vulns []csaf.Vulnerability, pids []csaf.ProductID) (map
+	case "fixed":
+		if p2.name == "" && p2.version == "" {
+			return nil, nil
+		}
+
+		if len(p2.archs) == 1 && p2.archs[0] == "src" {
+			return nil, nil
+		}
+
+		return &vcTypes.Criterion{
+			Vulnerable: true,
+			FixStatus:  &fixstatusTypes.FixStatus{Class: fixstatusTypes.ClassFixed},
+			Package: vcPackageTypes.Package{
+				Type: vcPackageTypes.PackageTypeBinary,
+				Binary: &vcBinaryPackageTypes.Package{
+					Name: func() string {
+						if p2.modularitylabel != "" {
+							return fmt.Sprintf("%s::%s", p2.modularitylabel, p2.name)
 						}
-						ss = append(ss, severityTypes.Severity{
-							Type:    severityTypes.SeverityTypeCVSSv30,
-							Source:  "secalert@redhat.com",
-							CVSSv30: v30,
-						})
-					case strings.HasPrefix(ass.severity.cvss3, "CVSS:3.1"):
-						v31, err := cvssV31Types.Parse(ass.severity.cvss3)
-						if err != nil {
-							return nil, errors.Wrap(err, "parse cvss3")
+						return p2.name
+					}(),
+					Architectures: slices.DeleteFunc(p2.archs, func(x string) bool { return x == "src" }),
+					Repositories:  p2.repositories,
+				},
+			},
+			Affected: &affectedTypes.Affected{
+				Type:  rangeTypes.RangeTypeRPM,
+				Range: []rangeTypes.Range{{LessThan: p2.version}},
+				Fixed: []string{p2.version},
+			},
+		}, nil
+	case "affected":
+		if p2.name == "" {
+			return nil, errors.New("name is empty")
+		}
+		if len(p2.archs) != 1 || p2.archs[0] != "src" {
+			return nil, errors.Errorf("unexpected affected pkg arch. expected: %q, actual: %q", []string{"src"}, p2.archs)
+		}
+
+		return &vcTypes.Criterion{
+			Vulnerable: true,
+			FixStatus: &fixstatusTypes.FixStatus{
+				Class:  fixstatusTypes.ClassUnfixed,
+				Vendor: ass.status.affected_status,
+			},
+			Package: vcPackageTypes.Package{
+				Type: vcPackageTypes.PackageTypeSource,
+				Source: &vcSourcePackageTypes.Package{
+					Name: func() string {
+						if p2.modularitylabel != "" {
+							return fmt.Sprintf("%s::%s", p2.modularitylabel, p2.name)
 						}
-						ss = append(ss, severityTypes.Severity{
-							Type:    severityTypes.SeverityTypeCVSSv31,
-							Source:  "secalert@redhat.com",
-							CVSSv31: v31,
-						})
-					default:
-						return nil, errors.Errorf("unexpected CVSSv3 string. expected: %q, actual: %q", "<score>/CVSS:3.[01]/<vector>", ass.severity.cvss3)
-					}
+						return p2.name
+					}(),
+					Repositories: p2.repositories,
+				},
+			},
+		}, nil
+	case "unaffected":
+		return nil, nil
+	default:
+		return nil, errors.Errorf("unexpected product_status. expected: %q, actual: %q", []string{"fixed", "affected", "unaffected"}, ass.status.product_status)
+	}
+}
+
+func buildVulnerabilities(baseVulnerabilities map[string]vulnerabilityContentTypes.Content, vassm map[string]map[csaf.ProductID]ass, pmmm map[string]map[ass]map[string]productsWithMaxPid) ([]vulnerabilityTypes.Vulnerability, error) {
+	// CVE ID -> severity -> major -> max Pid list
+	vpmmm := make(map[string]map[severity]map[string][]string)
```

Comment:
If the comment is correct, shouldn't it be `vsmmm` and not `vpmmm`?
```suggestion
	// CVE ID -> severity -> major -> max Pid list
	vsmmm := make(map[string]map[severity]map[string][]string)
```